### DISCUSSION TITLE
SD Card

### DISCUSSION THREAD

**Discussion by scytacki** [meatpiHQ/wican-fw#26] [General]
Has anyone tried wiring up an SD card to the WiCAN? 
I don't have one yet, so I'm not sure what pins from the ESP32 it exposes. I'm hoping it exposes some of the GPIO pins then I think I can use a board supporting SPI like this one: https://www.adafruit.com/product/4682

My goal is to use the WiCAN to do data logging while the car is away from the house and then when it gets back home it would upload the data. I'm assuming I need an SD card because the amount and frequency of data being recorded would be too much for the ESP's onboard flash, but I might be wrong about that.

**Comment by meatpiHQ** [comment]
@scytacki 

I wanted to prototype this addon board a while ago. But never had a chance. It plugs into the 2x5 connector.

![image](https://user-images.githubusercontent.com/94690098/225287831-75999d62-709a-4d6c-ac2c-a484e94a97b7.png)

You can use the exposed GPIOs in the 2x5 connector. You'll need to solder directly on the connector pins. Use IO1, IO2, IO5 and IO10. Don't use IO6 and IO4.

![image](https://user-images.githubusercontent.com/94690098/225282802-a38c3741-7ee7-4790-bb36-ca4d0095d0c1.png)
![image](https://user-images.githubusercontent.com/94690098/225283114-920632a4-224f-49b5-a7df-f41daa832574.png)

**Reply by scytacki** [reply]
Awesome! Thanks!

**Comment by js1tr3** [comment]
is there any SD card logging of can frames?

**Reply by meatpiHQ** [reply]
Yes on the Pro version, I'm still working on the firmware for that. The standard version does not have a sdcard connector.

**Reply by gytisrepecka** [reply]
Is logging to SD card implemented in Pro version?

I'm getting errors of unavailable JavaScript resources when I visit Dashboard page 🧐

**Reply by meatpiHQ** [reply]
Yes it's implemented. WiCAN tries to download some resources to the sdcard and it's failing because it cannot reach the internet. Usually this is due to subnet conflict. You should enable "Auto AP Disable" and try again.

**Reply by peregrines2** [reply]
Hi! I have another question regarding SD Card. How should it be formated? Or is WICAN PRO going to format it once introduced? Thank You :)

**Reply by meatpiHQ** [reply]
You should format the sdcard as fat32, and WiCAN will not format the card on it own.

**Reply by peregrines2** [reply]
Thank You!

**Reply by ialex87** [reply]
Sorry for the dumb question, is there a way to push data from scdard to mqtt based on event? Eg. If wifi is connected dump the sdcard logs to mqtt? Thanks!

**Reply by meatpiHQ** [reply]
It's doable. But not useful with home assistant because it does not support historical data
